keyboards/gmmk/gmmk2/p65/ansi/keymaps/vnmm/rgb_matrix_user.c

```diff
@@ -0,0 +1,94 @@
+/*
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include QMK_KEYBOARD_H
+#include "rgb_matrix_user.h"
+#include "keymap_user.h"
+
+keypos_t led_index_key_position[RGB_MATRIX_LED_COUNT];
+bool is_shift_pressed = false;
+
+void rgb_matrix_init_user(void) {
+    for (uint8_t row = 0; row < MATRIX_ROWS; row++) {
+        for (uint8_t col = 0; col < MATRIX_COLS; col++) {
+            uint8_t led_index = g_led_config.matrix_co[row][col];
+            if (led_index != NO_LED) {
+                led_index_key_position[led_index] = (keypos_t){.row = row, .col = col};
+            }
+        }
+    }
+}
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+    switch (keycode) {
+        case KC_LSFT:
+        case KC_RSFT:
+            is_shift_pressed = record->event.pressed;
```

Comment:
This wouldn't work correctly if you press one, then the other, and release either (variable would mark not pressed while one of them is actually pressed).

You could use `get_mods() & MOD_MASK_SHIFT`. If you 
* Remove `process_record_user`
* Replace the variable with a define (`#define is_shift_pressed (get_mods()  & MOD_MASK_SHIFT)

It *should* not break your existing code